hera_mc/cm_redis_corr.py

```diff
@@ -72,25 +36,35 @@ def cminfo_redis_snap(cminfo, redis_info=None):
     snap_to_ant = {}
     ant_to_snap = {}
     all_snap_inputs = {}
-    for antn, ant in enumerate(cminfo['antenna_numbers']):
-        name = cminfo['antenna_names'][antn]
-        pol_info = {}
-        for pol_snap in cminfo['correlator_inputs'][antn]:
-            pol = pol_snap.lower()[0]
-            if pol in ['e', 'n'] and '>' in pol_snap:
-                pol_info[pol] = pol_snap
+    snap_to_serial = {}
+    for _n, ant in enumerate(cminfo['antenna_numbers']):
+        name = cminfo['antenna_names'][_n]
+        corr_inp = cminfo['correlator_inputs'][_n]
+        snap_snr = cminfo['snap_serial_numbers'][_n]
         ant_to_snap[ant] = {}
-        for pol, psnap in pol_info.items():
-            snapi, channel = snap_part_to_host_input(psnap, redis_info=redis_info)
-            ant_to_snap[ant][pol] = {'host': snapi, 'channel': channel}
-            snap_to_ant.setdefault(snapi, [None] * 6)
-            snap_to_ant[snapi][channel] = name + pol.upper()
-            snap, adc_num = snap_part_to_host_input(psnap, None)
-            all_snap_inputs.setdefault(snap, [])
-            all_snap_inputs[snap].append(adc_num)
+        for _i, psnap in enumerate(corr_inp):
+            pol = psnap.lower()[0]
+            if pol not in ['e', 'n'] or '>' not in psnap:
+                warnings.warn(f"{psnap} is not an allowed correlator input")
+                continue
+            snap_input, snap_hostname = psnap.split('>')
+            channel = int(snap_input[1:]) // 2  # divide by 2 because ADC is in demux 2
+            ant_to_snap[ant][pol] = {'host': snap_hostname, 'channel': channel}
+            snap_to_ant.setdefault(snap_hostname, [None] * 6)
+            snap_to_ant[snap_hostname][channel] = name + pol.upper()
+            all_snap_inputs.setdefault(snap_hostname, [])
+            all_snap_inputs[snap_hostname].append(snap_input)
+            try:  # if already present make sure it agrees
+                if snap_to_serial[snap_hostname] != snap_snr[_i]:
+                    msg = "{} inconsistent: {}  !=  {}".format(
+                          snap_hostname, snap_to_serial[snap_hostname], snap_snr[_i])
+                    raise ValueError(msg)
```

Comment:
I'm definitely more liking this error raising. Can we maybe tweak the message to read more like
> The snap hostname {snap_hostname} differs between {type of input device}. {parameret 1} has value {one value}, {parameter 2} has value {two value} 
is it possible to even construct those parameter 1 and 2 parts? maybe just at least the first sentence and then follow up with what the two values are like what you have now.

@plaplant any additional thoughts on this?

Reply:
I've added the other piece of information we have at hand, which is the offending antpol.  Getting the other hostname would require another dictionary etc, which seems unnecessary since the fix is going into the database and fixing it and with the information provided in the error message you have enough information to find it.

Reply:
I think the point I was trying to make is the error message is very curt and more difficult to parse than the one you had before even.